x-pack/plugins/reporting/export_types/printable_pdf/server/execute_job/compatibility_shim.js

```diff
@@ -10,28 +10,28 @@ import { getAbsoluteUrlFactory } from './get_absolute_url';
 export function compatibilityShimFactory(server) {
   const getAbsoluteUrl = getAbsoluteUrlFactory(server);
 
-  const getSavedObjectAbsoluteUrl = (savedObj) => {
-    if (savedObj.urlHash) {
-      return getAbsoluteUrl({ hash: savedObj.urlHash });
+  const getSavedObjectAbsoluteUrl = (job, savedObject) => {
+    if (savedObject.urlHash) {
```

Comment:
Interesting, now that I see this... I guess reporting is supposed to work when storeStateInSessionStorage is turned on?  I have this bug: https://github.com/elastic/kibana/issues/20037 because there are issues.  But maybe it only fails when the url is too long, otherwise it can work with the setting.   Either way, we have no test coverage for this path.  I'd have to test manually to see if the new basePath/spaces stuff works with the setting turned on.

Reply:
Reporting should be expanding the hashes client-side before creating the job. The `urlHash` is an old version of the `relativeUrl` which doesn't include the `/app/kibana` prefix that we used to be able to rely upon until Canvas came around and needed to be able to target different "kibana applications".